warehouse/legacy/api/pypi.py

```diff
@@ -786,11 +807,12 @@ def file_upload(request):
         if filename.endswith(".whl"):
             wheel_info = _wheel_file_re.match(filename)
             plats = wheel_info.group("plat").split(".")
-            if set(plats) - ALLOWED_PLATFORMS:
-                raise _exc_with_message(
-                    HTTPBadRequest,
-                    "Binary wheel for an unsupported platform.",
-                )
+            for plat in plats:
+                if not _valid_platform_tag(plat):
+                    raise _exc_with_message(
+                        HTTPBadRequest,
+                        "Binary wheel for an unsupported platform.",
```

Comment:
I agree, though that would probably require building a dictionary that maps platform tag to Valid/Invalid and then doing an all() on the values of that dictionary or something.

@njsmith If you don't mind making this change in this PR, I think that would be a great addition.